crates/bevy_asset/src/server/mod.rs

```diff
@@ -1339,6 +1339,28 @@ pub enum LoadState {
     Failed(Box<AssetLoadError>),
 }
 
+impl LoadState {
+    /// Returns `true` if this instance is [`LoadState::Loading`]
+    pub fn is_loading(&self) -> bool {
+        matches!(self, Self::Loading)
+    }
+    /// Returns `true` if this instance is [`LoadState::Loaded`]
+    pub fn is_loaded(&self) -> bool {
+        matches!(self, Self::Loaded)
+    }
+    // NOTE: an `is_not_loaded` method is intentionally not included, as it may mislead some users
+    // into thinking it is complementary to `is_loaded`.
+    // `NotLoaded` is a very specific failure mode and in most cases it is not necessary to directly check for it.
+    // If this is necessary the `matches!` macro can be used instead of a helper method.
```

Comment:
Unrelated to this PR but it might be worth renaming the `NotLoaded` variant to `Idle` or something to help clarify the distinction on the enum itself.

Reply:
There might be a better name, but I like how `NotLoaded` avoids making any statements about the asset. If an asset handle is `NotLoaded`, it could mean that it hasn't started loading *yet*, or it could mean that there is no asset associated with the handle and never will be, or that the handle is invalid. A name like `Idle` implies that the asset does exist in some form which isn't necessarily true

Reply:
Oh good point, I hadn't considered that 🤔